lighthouse-core/audits/bootup-time.js

```diff
@@ -72,19 +72,55 @@ class BootupTime extends Audit {
     return result;
   }
 
+  /**
+   * Summarize JS costs by URL
+   * @param {Map<string, Object<string, number>>} executionTimings
+   * @param {LH.Audit.Context} context
+   * @return {{results: Object, totalBootupTime: number, extendedInfo: Object}}
+   */
+  static execCostsByURL(executionTimings, context) {
+    let totalBootupTime = 0;
+    const settings = context.settings || {};
+    const multiplier = settings.throttlingMethod === 'simulate' ?
+        settings.throttling.cpuSlowdownMultiplier : 1;
+
+    /** @type {Object<string, Object<string, number>>} */
+    const extendedInfo = {};
+
+    const results = Array.from(executionTimings).map(([url, groups]) => {
+      // Add up the totalBootupTime for all the taskGroups
+      for (const [name, value] of Object.entries(groups)) {
+        groups[name] = value * multiplier;
+      }
+
+      extendedInfo[url] = groups;
+
+      const scriptingTotal = groups[groupIdToName.scripting] || 0;
+      const parseCompileTotal = groups[groupIdToName.scriptParseCompile] || 0;
+      totalBootupTime += scriptingTotal + parseCompileTotal;
```

Comment:
is the only change moving this line out of the for loop? Might be easier to just do that :)